transport/http/inbound.go

```diff
@@ -182,16 +184,24 @@ func (i *Inbound) start() error {
 	return nil
 }
 
-// Stop the inbound, closing the listening socket.
+// Stop the inbound using Shutdown, but with a fixed timeout.
 func (i *Inbound) Stop() error {
-	return i.once.Stop(i.stop)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	return i.Shutdown(ctx)
 }
 
-func (i *Inbound) stop() error {
-	if i.server == nil {
-		return nil
-	}
-	return i.server.Stop()
+// Shutdown the inbound, closing the listening socket, closing idle
+// connections, and waiting for all pending calls to complete.
+func (i *Inbound) Shutdown(ctx context.Context) error {
```

Comment:
If we're going to expose this, we need to decide how this behaves when the
configurable shutdown timeout is also specified. Does context take precedence
over configured shutdown timeout, and if so, what if context doesn't have a
timeout at all?

If we're unsure about what the behavior should be, it would be preferable to
keep Shutdown unexported.

Reply:
Good point, will unexport for now